### DISCUSSION TITLE
[HELP] Get Line Number using the content of a Line as input

### DISCUSSION THREAD

**Discussion by FelipeRearden** [SilentVoid13/Templater#1248] [Help] [answered]
Hello friends !!!

I have a question about content from a **Specific Line**.

As we know, we can get the content from a specific line like this:
```
const editor = app.workspace.activeLeaf.view.editor;
const lnum = editor.getCursor().line;
const fnum = editor.getLine(lnum)
new Notice(fnum, 10000);
```

**MY QUESTION** is about the reverse. **Does anybody know how to get the LINE NUMBER using the content of a line as the input?**

**Example:**
● the content of line 5 is "Lorem Ipsum"
● I need a snippet to use `"Lorem Ipsum"` as input and get `5` as result.

My workflow: I am using `.match();` to filter `tp.file.content` in a suggester. After selecting the "content" from the suggester, I need the line number to use `app.workspace.activeLeaf.view.editor.setCursor()` to go to the desired line. 

Does anybody could help me ? 🙏🙏🙏
Thanks for reading this

**Comment by steven-serka** [comment, marked as answer]
Messing around in one of my random list files and using tp directly i found this works to output the current file line number of a given text:
`
<% tp.file.content.split("\n").indexOf("<search_text_here>") %>
`

I would imagine this would need to be expanded upon for larger file sizes, but in a pinch and shorter file lengths this is probably fine.

note: `.indexOf()` returns 0 on the first element, so you'd need to add one if that matters

**Reply by FelipeRearden** [reply]
Hello @serkae !!!!!

Thank you very much for the reply 🙏 🙏 🙏

Amazing idea !!!! It a fantastic workaround that worked perfect on my end. Thank you very much for the help.

I am gonna stress on large notes but I my initial tests, it is working extremely well. 

Again, thank you for helping me on this question. You are a legend 🥇

**Reply by steven-serka** [reply]
Happy to help, thanks for the idea!
